### PR TITLE
Fix/wizard admin menu priority 3

### PR DESCRIPTION
**Pull request by ronchambers**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The [original wizards code](https://github.com/Automattic/newspack-plugin/blob/489ce0fb7390f9fabc5956844ac7b437c2c701ac/includes/wizards/class-wizard.php#L52) in `trunk` gave us the ability to adjust the priority of a wizard's `add_page` callback.  Somewhere in the timeline of the `epic/ia` development, this ability was hard-code then removed around the time we were fixing the parent menu custom ordering.  In the end, after [attempt 1](https://github.com/Automattic/newspack-plugin/pull/3534) and [attempt 2](https://github.com/Automattic/newspack-plugin/pull/3535), and lots of discussion, testing and a huddle with @miguelpeixe - we've settled on this PR in order to reinstate the removed code.

Changes:
- The parent menu order variable was renamed to be more descriptive of what it does: `$parent_menu_order`.
- The reinstated variable for add page priority was give a descriptive name: `$add_page_priority`.
- The old/hard-coded `newspack` parent menu was changed to `$this->parent_menu` so wizards can control which parent menu they are added to.
- Some minimal changes were made to the advertising wizards and the dashboard.

### How to test the changes in this Pull Request:

1. Pull this branch on top of `epic/ia`.
2. Run `npm run build` if you haven't already for `epic/ia`.
3. Verify the wp-admin menus are displaying as expected.

parent-order

advertising

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/wizards/class-wizard.php [hunk 1]
```diff
@@ -54,11 +54,21 @@ abstract class Wizard {
...
+
+	/**
+	 * Priority for when 'add_page' callback fires. Default is WordPress's default hook priority of 10
+	 * Override this on a per-wizard basis if there is a need to adjust the order that 'add_page' fires.
+	 * Example: set priority < 10 so add_page is called before other pages. Or > 10 so it's called after other
+	 * pages or plugins are loaded.
+	 *
+	 * @var int.
+	 */
+	protected $add_page_priority = 10;
```
Comment (miguelpeixe) [inline includes/wizards/class-wizard.php]: I remember we thought of `submenu_priority` here, but I like this idea better. Following the same logic, the priority is applied to the `admin_menu` hook, so perhaps it should be called `admin_menu_priority`. WDYT? This is non-blocking, though.

**Review by miguelpeixe — APPROVED** [review]
Thank you for all the discussions and tackling this! It's a nice and comprehensive solution. I left a non-blocking suggestion there, feel free to merge if you disagree.

File: includes/wizards/class-wizard.php (hunk 1 above)
Comment (ronchambers) [reply includes/wizards/class-wizard.php]: Agreed.  I kept going back and forth...  `$admin_menu_priority` is closer to what I had in original PR.  Updated.

**Comment by ronchambers** [comment]
@jaredrethman FYI.  If you notice any issues with advertising/sponsors let me know.  Thanks!

**Pull request merged**